doc/decisions/1_in_discussion/change_tracking.md

```diff
@@ -124,15 +139,13 @@ Downsides of this approach:
 Use the `backendData->keys` for change tracking
 
 We already store which keys have been returned by `kdbGet` for each backend within KDB.
-Currently, however, this is not a deep duplication, as we are returning the keys from `backendData->keys` directly.
+Currently, however, this is not a deep copy, as we are returning the keys from `backendData->keys` directly.
```

Comment:
`backendData->keys` is an implementation detail. Is it the internal cache or something else?

Reply:
If we leave out this detail, then yes it would probably be the same as Alternative 1. This distinction was made before copy-on-write, as `backendData->keys` already stores the keys returned, whereas for Alternative 1, we'd have to have another duplication.

With copy-on-write, this distinction gets much more fuzzy.

Reply:
```suggestion
Currently, however, this is not a deep copy, as we are returning the internally stored `Key` instances directly.
```

We don't need to say where they are stored, just that the keys are returned and therefore can be changed externally.